### PR TITLE
[BugFix] Fix delete not compact when every level less than min_compact_segment_num

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

delete backtrace mechanism will select upper level which has only one rowset, but after add `min_compact_segment_num`. the level rowsets large than `min_compact_segment_num` will be compacted.
so that delete backtrace mechanism may be fail in some case.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/size_tiered_compaction_policy.cpp [hunk 1]
```diff
@@ -131,6 +131,11 @@ Status SizeTieredCompactionPolicy::_pick_rowsets_to_size_tiered_compact(bool for
         force_base_compaction = true;
     }
```
Comment (chaoyli) [inline be/src/storage/size_tiered_compaction_policy.cpp:133]: Sorry, I can not get the point what's the bug is?

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by meegoo** [comment]
run all

**Comment by wanpengfei-git** [comment]
@Mergify rebase

**Comment by wanpengfei-git** [comment]
@Mergify rebase

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**